multi.go

```diff
@@ -115,17 +115,29 @@ func (m *MultiFinder) Collector(name string) (Collector, error) {
 	return Collector{}, nil
 }
 
-func (m *MultiFinder) Find(query Query) ([]File, error) {
-	// TODO: only send the query to the appropriate finders
-	files := []File{}
-	for proj, f := range m.getFinders() {
-		pF, err := f.Find(query)
-		if err != nil {
-			return nil, fmt.Errorf("find failed for %s: %v", proj, err)
-		}
-		files = append(files, pF...)
+func (m *MultiFinder) Find(query Query) ([]BGPDump, error) {
+	dumps := []BGPDump{}
+
+	// Extract the project name from the query's collectors
+	if len(query.Collectors) == 0 {
+		return nil, fmt.Errorf("no collectors specified in query")
+	}
+	projectName := query.Collectors[0].Project.Name
+
+	// Get the appropriate finder for the specified project
+	finder, exists := m.getFinderByProject(projectName)
+	if !exists {
+		return nil, fmt.Errorf("unknown project: '%s'", projectName)
 	}
-	return files, nil
+
+	// Perform the search using the appropriate finder
+	dump, err := finder.Find(query)
+	if err != nil {
+		return nil, fmt.Errorf("find failed for %s: %v", projectName, err)
+	}
+	dumps = append(dumps, dump...)
```

Comment:
If you're not planning on supporting multiple finders then this is redundant I think.